### PR TITLE
[[ Bug 14316 ]] Allow infinite result in div, times and pow operations

### PR DESCRIPTION
**Pull request by livecodeali**
Before merging, we need to check the siginfo struct upon handling the SIGFPE signal to set `errno` more specifically, as otherwise we are conflating eg divide by zero and out of range errors.

### CODE REVIEW

File: engine/src/executionerrors.h [hunk 1]
```diff
@@ -2778,8 +2778,16 @@ enum Exec_errors
     EE_BAD_PERMISSION_NAME,
     
     // {EE-0910} Property: value is not a data
-    EE_PROPERTY_NOTADATA
+    EE_PROPERTY_NOTADATA,
     
+    // {EE-0911} Math: domain error
+    EE_MATH_DOMAIN,
```
Comment (runrevmark) [inline engine/src/executionerrors.h]: We are probably safe to re-use the first DOMAIN / RANGE / DIV BY ZERO code that exists, and just mark all others as unused... So EE_ACOS_DOMAIN -> EE_MATH_DOMAIN (numeric: domain error (undefined)); EE_DIVIDE_RANGE -> EE_MATH_RANGE (numeric: range error (overflow)) ; EE_DIVIDE_ZERO -> EE_MATH_ZERO (numeric: divide by zero).

Comment (runrevmark) [reply engine/src/executionerrors.h]: Codes which they replace should be changed to EE_UNUSED_XXXX - where XXXX is the error code (formatted with 0s).

**Comment by runrevmark** [comment]
@livecodeali Apart from the suggestion about re-using three existing error codes - this looks absolutely great.

File: docs/notes/bugfix-14316.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Don't throw error when infinity used innocuously in arithmetic expressions
```
Comment (runrevmark) [inline docs/notes/bugfix-14316.md:1]: The release note should probably describe the changes - i.e. when domain/range/div-by-zero are thrown and the unification of the error codes.

**Comment by livecode-vulcan** [comment]
😎 test success 20273023ac01c8dd223bc4d85e851371a38053f8

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/479)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/471)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/471)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/471)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/471)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1993)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/221)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/220)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2024)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2015)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2013)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2047)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1984)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 2027302

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 20273023ac01c8dd223bc4d85e851371a38053f8

**Pull request merged**